doc/developer-guides/hld/ivshmem-hld.rst

```diff
@@ -0,0 +1,93 @@
+.. _ivshmem-hld:
+
+ACRN Shared Memory Based Inter-VM Communication
+###############################################
+
+To support inter virtual machines communication based on shared memory
+mechanism, ACRN device model/hypervisor emulate a virtual PCI device
+(Called ivshmem device) to expose the base address and size of shared
+memory.
+
+Inter-VM Communication Overview
+*******************************
+
+.. figure:: images/ivshmem-image1.png
+   :align: center
+   :name: ivshmem-architecture-overview
+
+   ACRN shared memory based inter-vm communication architecture
+
+Two solutions of ivshmem device: dm-land and hypervisor-land. For dm-land,
```

Comment:
Are these two solutions two components of a single solution, or are these alternatives to each other?

I'm not a big fan of calling these `dm-land` and `hyervisor-land`, we've not used these in the past. I think I would prefer to use plain English to say if something runs in the Service VM or in the Hypervisor. @dbkinder , thoughts?

Reply:
> Are these two solutions two components of a single solution, or are these alternatives to each other?

Currently, they are alternatives. dm-land shared memory is allocated from Service VM, hypervisor-land shared memory is allocated from hypervisor.  In the future, when the hypervisor-land is implemented, we will check whether these two solutions can be merged into one solution.

Reply:
> > Are these two solutions two components of a single solution, or are these alternatives to each other?
> 
> Currently, they are alternatives. dm-land shared memory is allocated from Service VM, hypervisor-land shared memory is allocated from hypervisor. In the future, when the hypervisor-land is implemented, we will check whether these two solutions can be merged into one solution.

Thanks for the clarification!